In [1]:
# load packages

import pandas as pd
import numpy as np


pd.set_option("display.max_rows", 99)
pd.options.display.float_format = '{:,.3f}'.format

In [2]:
iowa = pd.read_csv('iowa_eda.csv')

In [3]:
iowa.head()

Date   City  Zip Code County  Item Number  \
0  2018-09-24  ADAIR     50002  ADAIR        86691   
1  2018-09-24  ADAIR     50002  ADAIR        25606   
2  2018-09-24  ADAIR     50002  ADAIR        37994   
3  2018-09-24  ADAIR     50002  ADAIR        36304   
4  2018-09-24  ADAIR     50002  ADAIR        26826   

                   Item Description  Bottle Volume (ml)  State Bottle Cost  \
0  Jack Daniels Tennessee Fire Mini                 500              9.060   
1       Seagrams 7 Crown Bl Whiskey                 750              7.000   
2                    Smirnoff 80prf                 375              4.750   
3                     Hawkeye Vodka                 375              1.860   
4     Jack Daniels Old #7 Black Lbl                 750             15.570   

   State Bottle Retail  Bottles Sold  ...  \
0               13.590             3  ...   
1               10.500            12  ...   
2                7.130             6  ...   
3                2.790            24  ...   
4               23.360            12  ...   

                            Store Name  Category Name      Category  \
0  Casey's general store #2521 / adair         WHISKY 1,012,400.000   
1                 Kum & go #76 / adair         WHISKY 1,012,400.000   
2  Casey's general store #2521 / adair          VODKA 1,032,230.000   
3                 Kum & go #76 / adair          VODKA 1,032,230.000   
4                 Kum & go #76 / adair         WHISKY 1,012,400.000   

                Vendor Name  Profit / Item Profit / Invoice  Profit / ml  \
0  BROWN-FORMAN CORPORATION          4.530           13.590        0.009   
1           DIAGEO AMERICAS          3.500           42.000        0.005   
2           DIAGEO AMERICAS          2.380           14.280        0.006   
3            LUXCO-ST LOUIS          0.930           22.320        0.002   
4  BROWN-FORMAN CORPORATION          7.790           93.480        0.010   

   Profit / Invoice / Liter  Retail Price / ml  Bottle Price Category  
0                     9.060              0.027                 Normal  
1                     4.667              0.014            Inexpensive  
2                     6.347              0.019            Inexpensive  
3                     2.480              0.007            Inexpensive  
4                    10.387              0.031                 Normal  

[5 rows x 22 columns]

In [4]:
to_do_list = list(iowa.columns)

In [5]:
def next_up():
    to_do_list.pop()
    print(to_do_list[-1])

    


### Objectives:

1.Are different bottle sizes distributed differently?

2.How has alcohol consumption changed over time?

3.How is liquor type distributed across Iowa?

### 1. Are Bottle sizes distributed differently?

In [6]:
# list of percentiles
percent = list(range(0,105,5))

In [7]:
# how do bottle sizes differ in popularity
np.percentile(iowa['Bottle Volume (ml)'], percent)

array([2.00e+01, 2.00e+02, 3.75e+02, 3.75e+02, 5.00e+02, 7.50e+02,
       7.50e+02, 7.50e+02, 7.50e+02, 7.50e+02, 7.50e+02, 7.50e+02,
       7.50e+02, 7.50e+02, 1.00e+03, 1.00e+03, 1.75e+03, 1.75e+03,
       1.75e+03, 1.75e+03, 3.78e+05])

<b> 750 is by far the most popular size. 1750 is second and both 375 and 1000 come in third

In [8]:
# first up
to_do_list[-1]

'Bottle Price Category'


### Bottle Price Category


In [9]:
# function to explore different features by a specific category
def explore(col):
    print(to_do_list[-1])
    return iowa.groupby([to_do_list[-1]])[col].describe()

In [10]:
explore('Bottle Volume (ml)')

Bottle Price Category


count    mean     std    min     25%     50%  \
Bottle Price Category                                                         
Expensive                102,458.000 344.619 348.169 50.000  50.000  50.000   
Inexpensive           13,084,290.000 859.658 452.851 50.000 750.000 750.000   
Normal                 9,590,484.000 960.749 840.853 20.000 750.000 750.000   
Very expensive            34,990.000 403.622 357.968 50.000  50.000 300.000   

                            75%         max  
Bottle Price Category                        
Expensive               750.000   2,400.000  
Inexpensive           1,000.000   3,600.000  
Normal                1,750.000 378,000.000  
Very expensive          750.000   3,000.000


### Observations:

1. Something is 378,000 ml?
2. Inexpensive is the most popular category

In [11]:
explore('State Bottle Retail')

Bottle Price Category


count    mean     std    min    25%     50%  \
Bottle Price Category                                                        
Expensive                102,458.000  62.426  26.285 31.670 44.300  45.000   
Inexpensive           13,084,290.000   9.285   3.483  0.290  6.630   9.750   
Normal                 9,590,474.000  22.307  14.770  2.750 16.490  20.000   
Very expensive            34,990.000 149.307 229.612 47.700 77.400 131.240   

                          75%        max  
Bottle Price Category                     
Expensive              83.990    323.640  
Inexpensive            11.990     15.210  
Normal                 26.850  9,702.000  
Very expensive        194.960 11,520.000


### Observation:

The bottle price category was partly built on bottle cost. The only thing that stands out is something normal priced cost \$ 9702

In [12]:
explore('Bottles Sold')

Bottle Price Category


count   mean    std   min   25%   50%    75%  \
Bottle Price Category                                                         
Expensive                102,458.000  2.764  4.369 1.000 1.000 2.000  3.000   
Inexpensive           13,084,290.000 11.468 27.848 1.000 3.000 6.000 12.000   
Normal                 9,590,484.000  9.392 30.686 1.000 2.000 6.000 12.000   
Very expensive            34,990.000  2.239  7.069 1.000 1.000 1.000  2.000   

                             max  
Bottle Price Category             
Expensive                300.000  
Inexpensive            7,116.000  
Normal                15,000.000  
Very expensive           600.000

### Observation

- Very Expensive bottles typically are sold in singles
- Expensive Bottles also are sold in smaller quantities
- Inexpensive and Normal come in packs

In [13]:
explore('Sale (Dollars)')

Bottle Price Category


count    mean       std    min    25%     50%  \
Bottle Price Category                                                          
Expensive                102,458.000 168.686   290.328 32.250 74.960  96.260   
Inexpensive           13,084,290.000  92.865   258.843  0.500 25.980  64.560   
Normal                 9,590,474.000 201.062   671.889  2.790 45.000 101.280   
Very expensive            34,990.000 330.098 1,291.214 47.700 90.000 154.980   

                          75%         max  
Bottle Price Category                      
Expensive             188.980  13,770.000  
Inexpensive           115.920  81,064.800  
Normal                207.960 279,557.280  
Very expensive        294.300 116,094.000

In [14]:
explore('Category Name')

Bottle Price Category


count unique              top     freq
Bottle Price Category                                           
Expensive                102072     18  FLAVORED WHISKY    48141
Inexpensive            13076069     22            VODKA  3286706
Normal                  9582400     23            VODKA  1360220
Very expensive            34919     20           SCOTCH     7483

- Scotch is very expensive
- Vodka is not
- flavored whisky is mildly expensive

In [15]:
explore('Item Description')

Bottle Price Category


count unique  \
Bottle Price Category                    
Expensive                102458    538   
Inexpensive            13084290   3756   
Normal                  9590484   6318   
Very expensive            34990    372   

                                                            top    freq  
Bottle Price Category                                                    
Expensive              Fireball Cinnamon Whiskey Mini Dispenser   47658  
Inexpensive                                       Hawkeye Vodka  507616  
Normal                                             Black Velvet  183447  
Very expensive           Fireball Cinnamon Whiskey Party Bucket    6911

- I learned that fireball mini dispenser and fireball party buckets exist.

In [16]:
explore('Store Name')

Bottle Price Category


count unique                           top   freq
Bottle Price Category                                                      
Expensive                102458   1972     Central city liquor, inc.   1690
Inexpensive            13084290   2535  Hy-vee #3 / bdi / des moines  95688
Normal                  9590484   2550  Hy-vee #3 / bdi / des moines  94251
Very expensive            34990   1714     Central city liquor, inc.    806

- Central city liquor, inc. caters to higher priced things.  Approach them if you need to begin selling expensive products

In [17]:
# what is going on at central city?
iowa[(iowa['Store Name'] == 'Central city liquor, inc.') & (iowa['Bottle Price Category'] == 'Very expensive')]

Date        City  Zip Code County  Item Number  \
5104076  2018-09-06  DES MOINES     50314   POLK        48122   
5105301  2018-09-18  DES MOINES     50314   POLK        48122   
5106060  2018-10-01  DES MOINES     50314   POLK        48122   
5106388  2018-10-01  DES MOINES     50314   POLK        49156   
5106407  2018-09-28  DES MOINES     50314   POLK         4381   
...             ...         ...       ...    ...          ...   
5466552  2016-04-29  DES MOINES     50314   POLK        48122   
5466629  2016-05-02  DES MOINES     50314   POLK       994113   
5467439  2016-05-09  DES MOINES     50314   POLK       904486   
5474949  2018-09-10  DES MOINES     50314   POLK         5329   
5474997  2018-09-06  DES MOINES     50314   POLK        49156   

                        Item Description  Bottle Volume (ml)  \
5104076               Hennessy XO Cognac                 750   
5105301               Hennessy XO Cognac                 750   
5106060               Hennessy XO Cognac                 750   
5106388        Remy Martin XO Excellence                 750   
5106407                    Balvenie 21YR                 750   
...                                  ...                 ...   
5466552               Hennessy XO Cognac                 750   
5466629              Hennessy Paradis HA                 750   
5467439  The Dalmore 1263 King Alexander                 750   
5474949              Johnnie Walker Blue                 750   
5474997        Remy Martin XO Excellence                 750   

         State Bottle Cost  State Bottle Retail  Bottles Sold  ...  \
5104076             99.990              149.990            12  ...   
5105301             99.990              149.990            12  ...   
5106060             99.990              149.990            12  ...   
5106388             99.940              149.910             6  ...   
5106407            121.220              181.830             1  ...   
...                    ...                  ...           ...  ...   
5466552             99.990              149.990            24  ...   
5466629            419.990              629.990             2  ...   
5467439            113.940              170.910             6  ...   
5474949            130.000              195.000             1  ...   
5474997             99.940              149.910             6  ...   

                        Store Name  Category Name      Category  \
5104076  Central city liquor, inc.         BRANDY 1,052,100.000   
5105301  Central city liquor, inc.         BRANDY 1,052,100.000   
5106060  Central city liquor, inc.         BRANDY 1,052,100.000   
5106388  Central city liquor, inc.         BRANDY 1,052,100.000   
5106407  Central city liquor, inc.         SCOTCH 1,012,210.000   
...                            ...            ...           ...   
5466552  Central city liquor, inc.         BRANDY 1,052,100.000   
5466629  Central city liquor, inc.         BRANDY 1,052,100.000   
5467439  Central city liquor, inc.         SCOTCH 1,012,210.000   
5474949  Central city liquor, inc.         SCOTCH 1,012,210.000   
5474997  Central city liquor, inc.         BRANDY 1,052,100.000   

                          Vendor Name  Profit / Item Profit / Invoice  \
5104076       MOET HENNESSY USA, INC.         50.000          600.000   
5105301       MOET HENNESSY USA, INC.         50.000          600.000   
5106060       MOET HENNESSY USA, INC.         50.000          600.000   
5106388            R�MY COINTREAU USA         49.970          299.820   
5106407  WILLIAM GRANT AND SONS, INC.         60.610           60.610   
...                               ...            ...              ...   
5466552       MOET HENNESSY USA, INC.         50.000        1,200.000   
5466629       MOET HENNESSY USA, INC.        210.000          420.000   
5467439          E AND J GALLO WINERY         56.970          341.820   
5474949               DIAGEO AMERICAS         65.000           65.000   
5474997         

- This store is in des moines, so all the top selling stores by price category are in des moines
- This store buys a lot of expensive scotch and brandy
- Tears of Llorona exists

In [18]:
next_up()

Retail Price / ml


In [19]:
# Retail Price / ml
explore('Category Name')

Retail Price / ml


count unique                top  freq
Retail Price / ml                                      
0.001                 1      1            LIQUEUR     1
0.003                15      1     BOURBON WHISKY    15
0.003               233      1            LIQUEUR   233
0.003              1655      1           SCHNAPPS  1655
0.003              5805      3         RYE WHISKY  4431
...                 ...    ...                ...   ...
4.788                 1      1  SPECIAL PACKAGING     1
9.741                 2      1             BRANDY     2
11.600                7      1     BOURBON WHISKY     7
12.000                4      1             WHISKY     4
15.360                1      1     BOURBON WHISKY     1

[5491 rows x 4 columns]

- This was just a lot. 
- Brandy is heavily represented at the bottom, or highest retail/ ml

In [20]:
next_up()

Profit / Invoice / Liter


In [21]:
next_up()

Profit / ml


In [22]:
# Profit / ml worst 100
explore('Category Name').head(100)

Profit / ml


count unique                top  freq
Profit / ml                                       
-0.087           1      1    CANADIAN WHISKY     1
-0.022           1      1           SCHNAPPS     1
-0.008           1      1      COCKTAILS RTD     1
-0.005           1      1  SPECIAL PACKAGING     1
-0.004           2      1  SPECIAL PACKAGING     2
...            ...    ...                ...   ...
0.002         2598      1              VODKA  2598
0.002        14182      3           SCHNAPPS  8910
0.002         1162      1           SCHNAPPS  1162
0.002         2976      3              VODKA  2156
0.002         9569      4      COCKTAILS RTD  8556

[100 rows x 4 columns]

- Special packaging not a good purchase
- Cocktails RTD thin profit margins

In [23]:
# Profit / ml best 100
explore('Category Name').tail(100)

Profit / ml


count unique                top freq
Profit / ml                                     
0.227           8      1          SPECIALTY    8
0.230          13      1            LIQUEUR   13
0.233           7      2             SCOTCH    5
0.234           3      1             WHISKY    3
0.237         110      1      COCKTAILS RTD  110
...           ...    ...                ...  ...
1.596           1      1  SPECIAL PACKAGING    1
3.247           2      1             BRANDY    2
3.867           7      1     BOURBON WHISKY    7
4.000           4      1             WHISKY    4
5.120           1      1     BOURBON WHISKY    1

[100 rows x 4 columns]

- whisky and brandy, however some special packaging

In [24]:
next_up()

Profit / Invoice


In [25]:
next_up()

Profit / Item


In [26]:
# profit/ item
explore('Item Description').head(100)

Profit / Item


count unique                                           top  \
Profit / Item                                                               
-16.810            1      1                                    99 Oranges   
-14.470            1      1  Margaritaville Skinny Island Punch Margarita   
-4.340             1      1                     Canadian Club Whisky Mini   
-4.000             1      1                      Bird Dog Peach w/2-50mls   
-3.490             2      1                 Kahlua w/Absolut 50ml & Glass   
...              ...    ...                                           ...   
1.020          11515      9                           Canadian Ltd Whisky   
1.030          11547      3                                 Hawkeye Vodka   
1.030          23961      3                           Canadian Ltd Whisky   
1.040          12404      3                                 Hawkeye Vodka   
1.050          28769      4                                     Five Star   

                freq  
Profit / Item         
-16.810            1  
-14.470            1  
-4.340             1  
-4.000             1  
-3.490             2  
...              ...  
1.020           6389  
1.030          10970  
1.030          23563  
1.040          12317  
1.050          19555  

[100 rows x 4 columns]

- this could give an idea of what products sell so poorly that distributors took a loss on them to get them out of their inventory

In [27]:
next_up()

Vendor Name


In [28]:
next_up()

Category


In [29]:
next_up()

Category Name


In [30]:
# most popular item by category
explore('Item Description')

Category Name


count unique  \
Category Name                                   
BOURBON WHISKY                 1400309    753   
BRANDY                         1198734    426   
CANADIAN WHISKY                2197312    274   
COCKTAILS RTD                   655518    367   
FLAVORED GIN                     11943     16   
FLAVORED RUM                    326540    196   
FLAVORED VODKA                 1131511    662   
FLAVORED WHISKY                 979396    138   
GIN                             761938    252   
HIGH BEER                           32      2   
IOWA LOCAL                        2937     20   
LIQUEUR                        1442946    930   
MEZCAL                            7544     70   
NEUTRAL GRAIN                    52958     23   
NEUTRAL GRAIN SPIRIT FLAVORED    42124     56   
RUM                            2265155    666   
RYE WHISKY                      122681    110   
SCHNAPPS                       1544325    503   
SCOTCH                          487698    534   
SPECIAL PACKAGING               322170   1508   
SPECIALTY                        26812    893   
TEQUILA                        1087101    607   
VODKA                          4648430   1008   
WHISKY                         2079346   1331   

                                                                 top    freq  
Category Name                                                                 
BOURBON WHISKY                                              Jim Beam  240434  
BRANDY                                                   Hennessy VS  167000  
CANADIAN WHISKY                                         Black Velvet  583803  
COCKTAILS RTD                   Jose Cuervo Authentic Lime Margarita   93631  
FLAVORED GIN                               Seagrams Lime Twisted Gin    8420  
FLAVORED RUM                                      Malibu Coconut Rum   84312  
FLAVORED VODKA                             Uv Blue (raspberry) Vodka  115718  
FLAVORED WHISKY                            Fireball Cinnamon Whiskey  298683  
GIN                                           Seagrams Extra Dry Gin  134215  
HIGH BEER                                       Sam Adams Utopias HA      29  
IOWA LOCAL                                         Templeton 6YR Rye    2676  
LIQUEUR                                         Jagermeister Liqueur  194580  
MEZCAL                                            Monte Alban Mezcal    2413  
NEUTRAL GRAIN                                      Everclear Alcohol   36797  
NEUTRAL GRAIN SPIRIT FLAVORED        Ole Smoky Salty Caramel Whiskey    8259  
RUM                                        Captain Morgan Spiced Rum  209461  
RYE WHISKY                                             Templeton Rye   41303  
SCHNAPPS                               Paramount Peppermint Schnapps  102214  
SCOTCH                                            Johnnie Walker Red   46400  
SPECIAL PACKAGING                                  Crown Royal Peach   11802  
SPECIALTY                              HA Weller Special Reserve 7YR    1575  
TEQUILA                        Jose Cuervo Especial Reposado Tequila  118579  
VODKA                                                  Hawkeye Vodka  507616  
WHISKY                                   Seagrams 7 Crown Bl Whiskey  179084

- I really enjoyed looking at this. not sure how usful the information is

In [31]:
# who sells/ buys the most per category
explore('Store Name')

Category Name


count unique  \
Category Name                                   
BOURBON WHISKY                 1400309   2496   
BRANDY                         1198734   2477   
CANADIAN WHISKY                2197312   2517   
COCKTAILS RTD                   655518   2289   
FLAVORED GIN                     11943    461   
FLAVORED RUM                    326540   2135   
FLAVORED VODKA                 1131511   2467   
FLAVORED WHISKY                 979396   2499   
GIN                             761938   2464   
HIGH BEER                           32     22   
IOWA LOCAL                        2937    410   
LIQUEUR                        1442946   2502   
MEZCAL                            7544    456   
NEUTRAL GRAIN                    52958   1325   
NEUTRAL GRAIN SPIRIT FLAVORED    42124   1237   
RUM                            2265155   2522   
RYE WHISKY                      122681   1805   
SCHNAPPS                       1544325   2480   
SCOTCH                          487698   1950   
SPECIAL PACKAGING               322170   2338   
SPECIALTY                        26812    788   
TEQUILA                        1087101   2502   
VODKA                          4648430   2532   
WHISKY                         2079346   2539   

                                                                 top   freq  
Category Name                                                                
BOURBON WHISKY                          Hy-vee #3 / bdi / des moines  11683  
BRANDY                                     Central city liquor, inc.  14726  
CANADIAN WHISKY                          Hy-vee food store / spencer   9437  
COCKTAILS RTD                           Hy-vee #3 food and drugstore   5261  
FLAVORED GIN                            Hy-vee #3 food and drugstore    174  
FLAVORED RUM                                          Central city 2   3808  
FLAVORED VODKA                          Hy-vee #3 / bdi / des moines  17230  
FLAVORED WHISKY                Hy-vee food store #2 / council bluffs   4021  
GIN                                     Hy-vee #3 / bdi / des moines   9423  
HIGH BEER                                  Central city liquor, inc.      3  
IOWA LOCAL                                           Hy-vee #4 / wdm     54  
LIQUEUR                                 Hy-vee #3 / bdi / des moines  20809  
MEZCAL                                  Hy-vee #3 / bdi / des moines    366  
NEUTRAL GRAIN                    Hy-vee wine and spirits / iowa city    511  
NEUTRAL GRAIN SPIRIT FLAVORED    Hy-vee food store #2 / state ankeny    538  
RUM                                     Hy-vee #3 / bdi / des moines  16131  
RYE WHISKY                              Hy-vee #3 / bdi / des moines   2284  
SCHNAPPS                                Hy-vee #3 / bdi / des moines  19769  
SCOTCH                                     Central city liquor, inc.   6879  
SPECIAL PACKAGING                       Hy-vee #3 / bdi / des moines   2647  
SPECIALTY                                  Central city liquor, inc.   1474  
TEQUILA                                 Hy-vee #3 / bdi / des moines  18301  
VODKA                                   Hy-vee #3 / bdi / des moines  31571  
WHISKY                                     Central city liquor, inc.  14049

- Nice work hy-vee

In [32]:
# city
explore('City')

Category Name


count unique           top    freq
Category Name                                                      
BOURBON WHISKY                 1400309    450    DES MOINES  107509
BRANDY                         1198734    446    DES MOINES  155463
CANADIAN WHISKY                2197312    450    DES MOINES  147807
COCKTAILS RTD                   655518    436  CEDAR RAPIDS   43559
FLAVORED GIN                     11943    140    DES MOINES    1384
FLAVORED RUM                    326540    438    DES MOINES   22474
FLAVORED VODKA                 1131511    449    DES MOINES  118372
FLAVORED WHISKY                 979396    449    DES MOINES   83980
GIN                             761938    447    DES MOINES   76185
HIGH BEER                           32     19    DES MOINES       5
IOWA LOCAL                        2937    153    DES MOINES     263
LIQUEUR                        1442946    450    DES MOINES  122643
MEZCAL                            7544    142    DES MOINES    1556
NEUTRAL GRAIN                    52958    330    DES MOINES    2869
NEUTRAL GRAIN SPIRIT FLAVORED    42124    344    DES MOINES    2657
RUM                            2265155    452    DES MOINES  169562
RYE WHISKY                      122681    395    DES MOINES   10084
SCHNAPPS                       1544325    447    DES MOINES   94667
SCOTCH                          487698    396    DES MOINES   46029
SPECIAL PACKAGING               322170    444    DES MOINES   23934
SPECIALTY                        26812    243    DES MOINES    4610
TEQUILA                        1087101    449    DES MOINES  127759
VODKA                          4648430    451    DES MOINES  447077
WHISKY                         2079346    451    DES MOINES  163990

- Hello cedar rapids being the dominate buyer in a category

In [33]:
next_up()

Store Name


In [34]:
explore('Item Description')

Store Name


count unique                        top  \
Store Name                                                                   
'da booze barn / west bend          2279    802               Black Velvet   
10th hole inn & suite / gift shop    240     86         Five O'Clock Vodka   
16th ave bp / cedar rapids           166     31  Fireball Cinnamon Whiskey   
1st ave bp / cedar rapids            142     25              Hawkeye Vodka   
1st stop beverage shop             11883   1151                Hennessy VS   
...                                  ...    ...                        ...   
Yesway store #1198 / swea city       516     91              Hawkeye Vodka   
Your private bar / urbandale         547    316       Titos Handmade Vodka   
Yr dollar stop                       705    107    Remy Martin Vsop Cognac   
Z's quickbreak                      8038    384         Five O'clock Vodka   
Zapf's pronto market                5704    485               Black Velvet   

                                  freq  
Store Name                              
'da booze barn / west bend          68  
10th hole inn & suite / gift shop   11  
16th ave bp / cedar rapids          21  
1st ave bp / cedar rapids           20  
1st stop beverage shop             360  
...                                ...  
Yesway store #1198 / swea city      88  
Your private bar / urbandale        27  
Yr dollar stop                      52  
Z's quickbreak                     470  
Zapf's pronto market               243  

[2552 rows x 4 columns]

- this was more interesting than I thought.
- Broadbent distillery buys a lot of bella limoncello? 
According to their website they produce bella limoncello.
- Hennessey vs was on this list way more than I expected

In [35]:
explore('Category Name')

Store Name


count unique              top  freq
Store Name                                                            
'da booze barn / west bend          2279     20  CANADIAN WHISKY   351
10th hole inn & suite / gift shop    240     18         SCHNAPPS    37
16th ave bp / cedar rapids           166     11  FLAVORED WHISKY    46
1st ave bp / cedar rapids            142     10            VODKA    44
1st stop beverage shop             11878     22            VODKA  2752
...                                  ...    ...              ...   ...
Yesway store #1198 / swea city       516     14  CANADIAN WHISKY   145
Your private bar / urbandale         547     21            VODKA    83
Yr dollar stop                       705     12           BRANDY   267
Z's quickbreak                      8037     17            VODKA  1930
Zapf's pronto market                5690     18            VODKA   834

[2552 rows x 4 columns]

In [36]:
explore('Profit / Item')

Store Name


count  mean   std   min   25%   50%  \
Store Name                                                                   
'da booze barn / west bend         2,279.000 7.040 5.096 0.450 4.170 5.490   
10th hole inn & suite / gift shop    240.000 5.209 2.565 1.170 3.205 4.530   
16th ave bp / cedar rapids           166.000 2.560 3.157 0.450 0.972 1.670   
1st ave bp / cedar rapids            142.000 1.744 2.563 0.450 0.900 1.540   
1st stop beverage shop            11,883.000 4.939 7.708 0.450 1.790 3.620   
...                                      ...   ...   ...   ...   ...   ...   
Yesway store #1198 / swea city       516.000 4.349 3.214 0.580 2.620 3.720   
Your private bar / urbandale         547.000 7.122 5.440 0.630 3.330 6.340   
Yr dollar stop                       705.000 4.203 3.047 0.490 1.730 3.450   
Z's quickbreak                     8,038.000 3.720 2.582 0.560 2.050 3.250   
Zapf's pronto market               5,704.000 4.797 2.289 0.620 3.150 4.340   

                                    75%     max  
Store Name                                       
'da booze barn / west bend        8.580  82.500  
10th hole inn & suite / gift shop 7.572  13.240  
16th ave bp / cedar rapids        2.500  15.000  
1st ave bp / cedar rapids         2.053  30.000  
1st stop beverage shop            5.790 625.000  
...                                 ...     ...  
Yesway store #1198 / swea city    5.230  34.750  
Your private bar / urbandale      9.450  57.120  
Yr dollar stop                    5.040  13.500  
Z's quickbreak                    4.620  65.000  
Zapf's pronto market              6.250  16.500  

[2552 rows x 8 columns]

- Only a few stores can average higher than \$10 profit per item. This might tell you which stores have higher prices or which stores get bigger case/ pack discounts

In [37]:
next_up()

Volume Sold (Liters)


In [38]:
explore('Store Name').tail(100)

Volume Sold (Liters)


count unique                             top freq
Volume Sold (Liters)                                                  
2,148.000                1      1  Sam's club 8162 / cedar rapids    1
2,152.500                1      1          Sam's club 6568 / ames    1
2,160.000                2      1   Hy-vee food store / urbandale    2
2,171.250                1      1   Hy-vee food store / urbandale    1
2,173.500                1      1     Costco wholesale #788 / wdm    1
...                    ...    ...                             ...  ...
11,340.000               1      1     Costco wholesale #788 / wdm    1
11,812.500               1      1     Costco wholesale #788 / wdm    1
11,880.000               1      1    Hy-vee #3 / bdi / des moines    1
13,200.000               3      1    Hy-vee #3 / bdi / des moines    3
15,000.000               2      1    Hy-vee #3 / bdi / des moines    2

[100 rows x 4 columns]

- that tells me who buys in bulk

In [39]:
explore('Store Name').head(100)

Volume Sold (Liters)


count unique                              top  freq
Volume Sold (Liters)                                                      
0.020                    417    233               J d spirits liquor    28
0.025                     41     31                 The liquor store     6
0.040                    268    138    Hy-vee food store / muscatine    24
0.050                 229091   1861    Hy-vee wine and spirits / wdm  1888
0.060                    135     80    Hy-vee food store / muscatine    16
...                      ...    ...                              ...   ...
3.700                     97     47        Pit stop liquors / newton     9
3.750                 148193   2027  Hy-vee food store / cedar falls  3787
3.800                     78     53      Quik trip #554 / sw 63rd dm     5
3.900                     42     22      Cork and bottle / oskaloosa     8
3.950                      1      1              Wal-mart 3762 / wdm     1

[100 rows x 4 columns]

- and who often gets small orders

In [40]:
next_up()

Sale (Dollars)


In [41]:
# i think this will be similiar to volume
explore('Store Name')

Sale (Dollars)


count unique                                       top freq
Sale (Dollars)                                                            
0.500              3      3              Fareway stores #531 / algona    1
1.300              2      2             Hy-vee wine and spirits / wdm    1
1.340            219    125              Hy-vee food store / johnston   12
1.350            334    210  Randy's neighborhood market / dyersville   12
1.430             58     37       Hy-vee / drugtown #1 / cedar rapids    7
...              ...    ...                                       ...  ...
196,004.880        1      1              Hy-vee #3 / bdi / des moines    1
225,838.800        1      1              Hy-vee #3 / bdi / des moines    1
250,932.000        3      1              Hy-vee #3 / bdi / des moines    3
254,100.000        2      1              Hy-vee #3 / bdi / des moines    2
279,557.280        1      1             Hy-vee food store / urbandale    1

[31120 rows x 4 columns]

In [42]:
next_up()

Bottles Sold


In [43]:
next_up()

State Bottle Retail


In [44]:
explore('Category Name')

State Bottle Retail


count unique             top freq
State Bottle Retail                                  
0.290                   1      1        SCHNAPPS    1
0.500                  15      1  BOURBON WHISKY   15
0.650                   7      2         LIQUEUR    6
0.720                   1      1         LIQUEUR    1
0.760                   1      1        SCHNAPPS    1
...                   ...    ...             ...  ...
8,700.000               7      1  BOURBON WHISKY    7
9,000.000               4      1          WHISKY    4
9,150.000               2      1          WHISKY    2
9,702.000               2      1          WHISKY    2
11,520.000              1      1  BOURBON WHISKY    1

[3687 rows x 4 columns]

In [45]:
next_up()

State Bottle Cost


In [46]:
next_up()

Bottle Volume (ml)


In [47]:
next_up()

Item Description


In [48]:
explore('Profit / Item')

Item Description


count   mean   std    min    25%  \
Item Description                                                               
"Beefeater ""24"""                          2.000  6.250 0.000  6.250  6.250   
"Jose Cuervo Silver ""Rolling Stones"""   194.000  5.250 0.000  5.250  5.250   
"Rumchata ""GoChatas"""                     3.000 49.500 0.000 49.500 49.500   
"Rumchata ""MiniChatas"" Creamer Cups"  1,352.000  3.500 0.000  3.500  3.500   
(ri)1 Rye                                 197.000 10.500 0.000 10.500 10.500   
...                                           ...    ...   ...    ...    ...   
Zubrowka Bison Grass Vodka                 32.000  6.490 0.000  6.490  6.490   
Zwack                                      21.000  7.350 0.799  6.240  6.430   
Zwack Unicum                                3.000  7.960 0.000  7.960  7.960   
Zyr Vodka                                 147.000  8.962 2.282  7.500  8.440   
maDIKwe                                   216.000  5.710 0.331  4.500  5.800   

                                           50%    75%    max  
Item Description                                              
"Beefeater ""24"""                       6.250  6.250  6.250  
"Jose Cuervo Silver ""Rolling Stones"""  5.250  5.250  5.250  
"Rumchata ""GoChatas"""                 49.500 49.500 49.500  
"Rumchata ""MiniChatas"" Creamer Cups"   3.500  3.500  3.500  
(ri)1 Rye                               10.500 10.500 10.500  
...                                        ...    ...    ...  
Zubrowka Bison Grass Vodka               6.490  6.490  6.490  
Zwack                                    7.960  7.960  7.960  
Zwack Unicum                             7.960  7.960  7.960  
Zyr Vodka                                8.440  8.440 16.880  
maDIKwe                                  5.800  5.800  5.800  

[10082 rows x 8 columns]

In [49]:
explore('City')

Item Description


count unique           top freq
Item Description                                                       
"Beefeater ""24"""                          2      2          AMES    1
"Jose Cuervo Silver ""Rolling Stones"""   194     47  CEDAR RAPIDS   27
"Rumchata ""GoChatas"""                     3      2  CEDAR RAPIDS    2
"Rumchata ""MiniChatas"" Creamer Cups"   1352    133          AMES   71
(ri)1 Rye                                 197     39    DES MOINES   53
...                                       ...    ...           ...  ...
Zubrowka Bison Grass Vodka                 32      6    DES MOINES    9
Zwack                                      21      7    DES MOINES   10
Zwack Unicum                                3      3    DES MOINES    1
Zyr Vodka                                 147     22    DES MOINES   35
maDIKwe                                   216     29    DES MOINES   69

[10082 rows x 4 columns]

In [50]:
next_up()

Item Number


In [51]:
next_up()

County


In [52]:
explore('Bottle Price Category')

County


count unique          top     freq
County                                             
ADAIR            46906      4  Inexpensive    28904
ADAMS            18664      4  Inexpensive    10025
ALLAMAKEE        94728      4  Inexpensive    54186
APPANOOSE        82274      4  Inexpensive    49606
AUDUBON          24637      4  Inexpensive    15539
BENTON          105664      4  Inexpensive    63946
BLACK HAWK     1281301      4  Inexpensive   746211
BOONE           164621      4  Inexpensive    96672
BREMER          178119      4  Inexpensive   103652
BUCHANAN        129311      4  Inexpensive    75219
BUENA VISTA     196842      4  Inexpensive   110580
BUTLER           38348      4  Inexpensive    25552
CALHOUN          34765      4  Inexpensive    23472
CARROLL         163272      4  Inexpensive    92537
CASS            103000      4  Inexpensive    59279
CEDAR            90552      4  Inexpensive    61057
CERRO GORDO     507822      4  Inexpensive   287912
CHEROKEE         78810      4  Inexpensive    45495
CHICKASAW        36533      4  Inexpensive    23146
CLARKE           58141      4  Inexpensive    33808
CLAY            160682      4  Inexpensive    93713
CLAYTON         101177      4  Inexpensive    57392
CLINTON         308546      4  Inexpensive   181061
CRAWFORD         92369      4  Inexpensive    52425
DALLAS          254942      4  Inexpensive   140663
DAVIS            17149      4  Inexpensive    11789
DECATUR          20486      4  Inexpensive    12809
DELAWARE         69246      4  Inexpensive    38005
DES MOINES      343043      4  Inexpensive   205144
DICKINSON       290647      4  Inexpensive   156923
DUBUQUE         667621      4  Inexpensive   382393
EMMET            57290      4  Inexpensive    34878
FAYETTE          97203      4  Inexpensive    58933
FLOYD            91908      4  Inexpensive    53218
FRANKLIN         62270      4  Inexpensive    35983
FREMONT           3567      4  Inexpensive     2242
GREENE           51784      4  Inexpensive    31210
GRUNDY           43235      4  Inexpensive    28812
GUTHRIE          44164      4  Inexpensive    27195
HAMILTON        106479      4  Inexpensive    65881
HANCOCK          34917      4  Inexpensive    21860
HARDIN          141582      4  Inexpensive    82787
HARRISON         90115      4  Inexpensive    54013
HENRY            88849      4  Inexpensive    50427
HOWARD           63811      4  Inexpensive    36334
HUMBOLDT         54526      4  Inexpensive    32243
IDA              52760      4  Inexpensive    28564
IOWA            139482      4  Inexpensive    85484
JACKSON         126452      4  Inexpensive    78553
JASPER          211386      4  Inexpensive   126288
JEFFERSON        80139      4  Inexpensive    46238
JOHNSON        1149125      4  Inexpensive   618667
JONES           150784      4  Inexpensive    89569
KEOKUK           37255      4  Inexpensive    24506
KOSSUTH         133084      4  Inexpensive    73736
LEE             268144      4  Inexpensive   153353
LINN           1901012      4  Inexpensive  1111070
LOUISA           42084      4  Inexpensive    28397
LUCAS            49332      4  Inexpensive    29479
LYON             93063      4  Inexpensive    51769
MADISON          79153      4  Inexpensive    45103
MAHASKA         113523      4  Inexpensive    61239
MARION          216964      4  Inexpensive   125575
MARSHALL        260705      4  Inexpensive   147633
MILLS            60432      4  Inexpensive    33637
MITCHELL         74154      4  Inexpensive    42947
MONONA          102549      4  Inexpensive    59704
MONROE           31048      4  Inexpensive    18746
MONTGOMERY       62044      4  Inexpensive    35206
MUSCATINE       337694      4  Inexpensive   201163
O'BRIEN         153797      4  Inexpensive    90224
OSCEOLA          32647      4  Inexpensive    18534
PAGE            114478      4  Inexpensive    66035
PALO ALTO        95554      4  Inexpensive    53516
PLYMOUTH        131324      4  Inexpensive    71701
POCAHONTAS   

<b> Was not sure, if this result was going to be the same. Thought maybe we could see Normal

In [53]:
explore('Category Name')

County


count unique              top    freq
County                                                
ADAIR            46894     22            VODKA    9774
ADAMS            18651     21  CANADIAN WHISKY    3231
ALLAMAKEE        94701     23            VODKA   16924
APPANOOSE        82169     23            VODKA   14248
AUDUBON          24636     22            VODKA    4742
BENTON          105576     23            VODKA   19395
BLACK HAWK     1280054     24            VODKA  254082
BOONE           164480     23            VODKA   31160
BREMER          177967     23            VODKA   30437
BUCHANAN        129221     23            VODKA   23693
BUENA VISTA     196735     24            VODKA   33159
BUTLER           38345     20            VODKA    8897
CALHOUN          34765     20            VODKA    7689
CARROLL         163178     23            VODKA   30213
CASS            102927     23            VODKA   18688
CEDAR            90517     22            VODKA   17931
CERRO GORDO     507473     23            VODKA   98737
CHEROKEE         78739     23            VODKA   13639
CHICKASAW        36513     21            VODKA    6842
CLARKE           58100     22            VODKA   10346
CLAY            160545     23            VODKA   28868
CLAYTON         101094     23            VODKA   16502
CLINTON         308253     23            VODKA   62277
CRAWFORD         92279     23            VODKA   13835
DALLAS          254794     23            VODKA   57146
DAVIS            17133     20            VODKA    3449
DECATUR          20485     20            VODKA    4283
DELAWARE         69204     22            VODKA   12622
DES MOINES      342842     23            VODKA   75728
DICKINSON       290499     23            VODKA   59170
DUBUQUE         667343     23            VODKA  139591
EMMET            57266     22            VODKA    9843
FAYETTE          97109     23            VODKA   17159
FLOYD            91816     23            VODKA   18304
FRANKLIN         62248     23            VODKA   10583
FREMONT           3567     19            VODKA     815
GREENE           51775     22            VODKA    9556
GRUNDY           43223     22            VODKA   11328
GUTHRIE          44155     21            VODKA    9688
HAMILTON        106426     23            VODKA   20561
HANCOCK          34917     23            VODKA    7730
HARDIN          141461     23            VODKA   30441
HARRISON         90044     23            VODKA   13743
HENRY            88778     24            VODKA   18066
HOWARD           63772     23            VODKA   11543
HUMBOLDT         54497     23            VODKA    9945
IDA              52681     24            VODKA    8554
IOWA            139416     23            VODKA   28359
JACKSON         126362     23            VODKA   22696
JASPER          211243     24            VODKA   39308
JEFFERSON        80037     24            VODKA   16760
JOHNSON        1148083     24            VODKA  260142
JONES           150660     23            VODKA   26491
KEOKUK           37246     22            VODKA    5935
KOSSUTH         132963     23            VODKA   22692
LEE             267909     23            VODKA   47698
LINN           1899966     24            VODKA  424005
LOUISA           42081     22            VODKA    6797
LUCAS            49285     23            VODKA    9050
LYON             92977     23  CANADIAN WHISKY   16889
MADISON          79134     23            VODKA   15592
MAHASKA         113410     23            VODKA   22260
MARION          216773     24            VODKA   42921
MARSHALL        260484     23            VODKA   49917
MILLS            60334     22            VODKA   12404
MITCHELL         74134     23            VODKA   13521
MONONA          102495     23            VODKA   16045
MONROE           31033     21            VODKA    5908
MONTGOMERY       61965     23            VODKA   11128
MUSCATINE       337461     23            VODKA   62882
O'BRIEN         153626     24  CANADIAN WHISKY   23138
OS

<b> Proof vodka pays the bills

In [54]:
explore('Store Name')

County


count unique                                          top  \
County                                                                       
ADAIR            46906     12                      Hometown foods / stuart   
ADAMS            18664      3                            West side spirits   
ALLAMAKEE        94728     15                 Quillins food ranch / waukon   
APPANOOSE        82274     11              Hy-vee food store / centerville   
AUDUBON          24637      5                            Audubon food land   
BENTON          105664     17                               Roy's foodland   
BLACK HAWK     1281301    147              Hy-vee food store / cedar falls   
BOONE           164621     23              Hy-vee wine and spirits / boone   
BREMER          178119     17                             Hy-vee / waverly   
BUCHANAN        129311     22               Hartig drug #14 / independence   
BUENA VISTA     196842     29         Hy-vee wine and spirits / storm lake   
BUTLER           38348     12                        Brothers market, inc.   
CALHOUN          34765     16                        Lake city food center   
CARROLL         163272     22                  Hy-vee food store / carroll   
CASS            103000     15           Hy-vee wine and spirits / atlantic   
CEDAR            90552     15                          Tipton family foods   
CERRO GORDO     507822     40            Hy-vee food store #1 / mason city   
CHEROKEE         78810     12                 Hy-vee food store / cherokee   
CHICKASAW        36533     10            Fareway stores #922 / new hampton   
CLARKE           58141     12                  Hy-vee food store / osceola   
CLAY            160682     16                  Hy-vee food store / spencer   
CLAYTON         101177     22                Franklin street floral & gift   
CLINTON         308546     38               Hy-vee food and drug / clinton   
CRAWFORD         92369     10            Hy-vee wine and spirits / denison   
DALLAS          254942     44                              Hy-vee / waukee   
DAVIS            17149      3                                Keith's foods   
DECATUR          20486      6                         Kum & go #4 / lamoni   
DELAWARE         69246     12                  Cork 'n bottle / manchester   
DES MOINES      343043     36                    Hy-vee #1044 / burlington   
DICKINSON       290647     22                        Okoboji avenue liquor   
DUBUQUE         667621     77                  Hy-vee food store / dubuque   
EMMET            57290      7        Hy-vee wine and spirits / estherville   
FAYETTE          97203     23                Fareway stores #412 / oelwein   
FLOYD            91908     17                        Hy-vee / charles city   
FRANKLIN         62270     14                 Northside one stop / hampton   
FREMONT           3567      3        Casey's general store #1330 / hamburg   
GREENE           51784      7                         Hy-vee  /  jefferson   
GRUNDY           43235     11  Casey's general store #3055 / grundy center   
GUTHRIE          44164      9                      Hometown foods / panora   
HAMILTON        106479     15             Hy-vee food store / webster city   
HANCOCK          34917      7         Casey's general store #3031 / garner   
HARDIN          141582     22               Hy-vee food store / iowa falls   
HARRISON         90115     11     Food land super markets / missouri valey   
HENRY            88849     12           Hy-vee food store / mount pleasant   
HOWARD           63811     12              The ox & wren spirits and gifts   
HUMBOLDT         54526      8           Hy-vee wine and spirits / humboldt   
IDA              52760      7                   Spirits, stogies and stuff   
IOWA            139482     27                             Big g food store   
JACKSON         126452     15                   B and c liquor / maquoketa   
JASPER          211386     28          

<b> the unique column is of most interest. How many retailers there are per county

In [55]:
explore('Item Description')

County


count unique                 top   freq
County                                                  
ADAIR            46906   1048        Black Velvet   2947
ADAMS            18664   1334        Black Velvet    782
ALLAMAKEE        94728   2164        Black Velvet   2848
APPANOOSE        82274   2479       Hawkeye Vodka   2958
AUDUBON          24637   1116        Black Velvet   1345
BENTON          105664   2783        Black Velvet   4315
BLACK HAWK     1281301   6528        Black Velvet  28049
BOONE           164621   3602        Black Velvet   5641
BREMER          178119   3696        Black Velvet   5037
BUCHANAN        129311   2651        Black Velvet   4136
BUENA VISTA     196842   3374        Black Velvet   4888
BUTLER           38348    919        Black Velvet   2816
CALHOUN          34765    803        Black Velvet   2875
CARROLL         163272   2893        Black Velvet   5183
CASS            103000   2526        Black Velvet   3752
CEDAR            90552   1507        Black Velvet   3779
CERRO GORDO     507822   5436        Black Velvet  11887
CHEROKEE         78810   2155        Black Velvet   2735
CHICKASAW        36533   1302        Black Velvet   1829
CLARKE           58141   1674        Black Velvet   2183
CLAY            160682   3270        Black Velvet   3367
CLAYTON         101177   2697        Black Velvet   4461
CLINTON         308546   4186        Black Velvet   5755
CRAWFORD         92369   2957        Black Velvet   3288
DALLAS          254942   4747       Hawkeye Vodka   8843
DAVIS            17149    594        Black Velvet    650
DECATUR          20486    674        Black Velvet   1219
DELAWARE         69246   2209        Black Velvet   2267
DES MOINES      343043   3488  Five O'clock Vodka   6777
DICKINSON       290647   4466        Black Velvet   6857
DUBUQUE         667621   4731        Black Velvet  14586
EMMET            57290   1999        Black Velvet   1526
FAYETTE          97203   2678        Black Velvet   3870
FLOYD            91908   2971        Black Velvet   2381
FRANKLIN         62270   2277        Black Velvet   2481
FREMONT           3567    332        Black Velvet    344
GREENE           51784   1899        Black Velvet   1949
GRUNDY           43235   1079        Black Velvet   2577
GUTHRIE          44164   1154        Black Velvet   2282
HAMILTON        106479   2251        Black Velvet   4044
HANCOCK          34917   1036        Black Velvet   2014
HARDIN          141582   2953        Black Velvet   5591
HARRISON         90115   2760        Black Velvet   2830
HENRY            88849   2579       Hawkeye Vodka   2830
HOWARD           63811   2361        Black Velvet   1609
HUMBOLDT         54526   1947        Black Velvet   1723
IDA              52760   2908        Black Velvet   1718
IOWA            139482   3293       Hawkeye Vodka   4332
JACKSON         126452   2140        Black Velvet   3530
JASPER          211386   3802        Black Velvet   7134
JEFFERSON        80139   2946       Hawkeye Vodka   2607
JOHNSON        1149125   6992       Hawkeye Vodka  26779
JONES           150784   2881        Black Velvet   4293
KEOKUK           37255   1768        Black Velvet   1318
KOSSUTH         133084   4477        Black Velvet   3594
LEE             268144   4088       Hawkeye Vodka   7805
LINN           1901012   6848       Hawkeye Vodka  48019
LOUISA           42084   1219        Black Velvet   2497
LUCAS            49332   1889        Black Velvet   1517
LYON             93063   2579        Black Velvet   3377
MADISON          79153   2442        Black Velvet   3291
MAHASKA         113523   3765        Black Velvet   3115
MARION          216964   3998        Black Velvet   5895
MARSHALL        260705   4678        Black Velvet   7428
MILLS            60432   2029        Black Velvet   2111
MITCHELL         74154   1618        Black Velvet   2529
MONONA          102549   2980        Black Velvet   2708
MONROE           31048   1109       Hawkeye Vodka   1471
MONTGOMERY       62044   24

<b> single item popularity is mostly black velvet (canadian whisky) but most popular category is vodka.
also, this unique column tell us how many different products are sold in each county

In [56]:
next_up()

Zip Code


In [57]:
explore('Category Name')

Zip Code


count unique                top   freq
Zip Code                                         
50002       5443     19              VODKA   1564
50003      31371     23              VODKA   6719
50005        855     15              VODKA    254
50006       9523     20    CANADIAN WHISKY   2008
50008        444     18  SPECIAL PACKAGING    155
...          ...    ...                ...    ...
52804     276374     23              VODKA  55218
52806     173358     23              VODKA  36274
52807     307674     23              VODKA  58568
56201       1686     17              VODKA    662
57222        902     16              VODKA    287

[496 rows x 4 columns]

<b> A little more diversity here, compared to county

In [58]:
explore('Item Description')

Zip Code


count unique                         top  freq
Zip Code                                                 
50002       5443    311                Black Velvet   471
50003      31391   1988               Hawkeye Vodka  1522
50005        855    137                Black Velvet    90
50006       9525    558                Black Velvet   697
50008        445    267  Weller Reserve 7YR Bourbon    14
...          ...    ...                         ...   ...
52804     276699   4443                 Hennessy VS  3938
52806     173480   3129               Hawkeye Vodka  2682
52807     307961   4828               Hawkeye Vodka  3362
56201       1688    162               Hawkeye Vodka   264
57222        902    144   Fireball Cinnamon Whiskey    65

[496 rows x 4 columns]

In [59]:
next_up()

City


In [60]:
explore('Category Name')

City


count unique              top  freq
City                                         
ACKLEY     6742     19            VODKA  1734
ADAIR      5443     19            VODKA  1564
ADEL      31371     23            VODKA  6719
AFTON      2727     17  CANADIAN WHISKY   683
AKRON      4844     17            VODKA  1039
...         ...    ...              ...   ...
WINTHROP   7239     20            VODKA  1258
WOODBINE  14510     20  CANADIAN WHISKY  2439
WOODWARD   3793     19            VODKA  1091
WYOMING     440     15            VODKA   133
ZWINGLE    2878     19            VODKA   689

[452 rows x 4 columns]

In [61]:
explore('Item Description')

City


count unique            top  freq
City                                       
ACKLEY     6742    411   Black Velvet   644
ADAIR      5443    311   Black Velvet   471
ADEL      31391   1988  Hawkeye Vodka  1522
AFTON      2727    200   Black Velvet   375
AKRON      4844    152   Black Velvet   478
...         ...    ...            ...   ...
WINTHROP   7239    397   Black Velvet   506
WOODBINE  14511    899   Black Velvet   579
WOODWARD   3793    209   Black Velvet   418
WYOMING     440    118   Black Velvet    35
ZWINGLE    2878    360   Black Velvet   116

[452 rows x 4 columns]

<b> city didnt offer much differencet than zip

In [62]:
next_up()

Date


In [63]:
explore('Vendor Name')

Date


count unique              top  freq
Date                                           
2012-01-03  10121     50  DIAGEO AMERICAS  1731
2012-01-04   9263     48  DIAGEO AMERICAS  1524
2012-01-05   8795     50  DIAGEO AMERICAS  1543
2012-01-09   9134     52  DIAGEO AMERICAS  1539
2012-01-10   8702     52  DIAGEO AMERICAS  1496
...           ...    ...              ...   ...
2021-12-27   9919    103  DIAGEO AMERICAS  1423
2021-12-28  13408    119  DIAGEO AMERICAS  2074
2021-12-29   8919     98  DIAGEO AMERICAS  1323
2021-12-30  10032     99  DIAGEO AMERICAS  1528
2021-12-31  10697     91  DIAGEO AMERICAS  1707

[2502 rows x 4 columns]

<b> This was a lot more interesting than I thought. Diageo, the world largest liquor company has most the orders for the first few years of data. after July 7th 2021 sazerac holds about half of most common invoice per date.
    
Did sazerac take over distribution of a major product like black velvet?
    
For future, model predictions of invoice market share by vendor

In [64]:
explore('Category Name')

Date


count unique    top  freq
Date                                 
2012-01-03  10121     18  VODKA  1743
2012-01-04   9262     18  VODKA  1534
2012-01-05   8795     18  VODKA  1745
2012-01-09   9134     18  VODKA  1659
2012-01-10   8702     18  VODKA  1650
...           ...    ...    ...   ...
2021-12-27   9919     21  VODKA  1738
2021-12-28  13408     21  VODKA  2259
2021-12-29   8919     21  VODKA  1612
2021-12-30  10032     21  VODKA  1959
2021-12-31  10697     20  VODKA  2132

[2502 rows x 4 columns]

In [65]:
explore('Item Description')

Date


count unique                   top freq
Date                                               
2012-01-03  10121    826          Black Velvet  182
2012-01-04   9263    845          Black Velvet  188
2012-01-05   8795    835    Five O'clock Vodka  152
2012-01-09   9134    838          Black Velvet  203
2012-01-10   8702    803          Black Velvet  223
...           ...    ...                   ...  ...
2021-12-27   9919   1284  Titos Handmade Vodka  260
2021-12-28  13408   1345  Titos Handmade Vodka  322
2021-12-29   8919   1285  Titos Handmade Vodka  203
2021-12-30  10032   1282  Titos Handmade Vodka  257
2021-12-31  10697   1109          Black Velvet  312

[2502 rows x 4 columns]

<b> Titos! became really popular in 2019 and seems to have only grown in popularity, perhaps titos is who changed distributer

In [97]:
iowa.to_csv('iowa_eda.csv', index=False)